Mobl_ios_Callisto-master/MealTrackingPilot/LogCommonMeal/Views/OccasionPicker.swift

```diff
@@ -693,93 +576,90 @@ fileprivate extension CGPoint {
 }
 
 fileprivate extension CGSize {
-    init(span: CGFloat, lateralSpan: CGFloat = 0, direction: PickerView.ScrollingDirection) {
+    init(span: CGFloat, lateralSpan: CGFloat = 0, direction: OccasionPicker.ScrollingDirection) {
         let width = direction == .horizontal ? span : lateralSpan
         let height = direction == .vertical ? span : lateralSpan
         self.init(width: width, height: height)
     }
-
-    func span(forDirection direction: PickerView.ScrollingDirection) -> CGFloat {
+    
+    func span(forDirection direction: OccasionPicker.ScrollingDirection) -> CGFloat {
         return direction == .horizontal ? width : height
     }
-
-    func lateralSpan(forDirection direction: PickerView.ScrollingDirection) -> CGFloat {
+    
+    func lateralSpan(forDirection direction: OccasionPicker.ScrollingDirection) -> CGFloat {
         return span(forDirection: direction.opposite())
     }
 }
 
-extension PickerView: UICollectionViewDelegate {
```

Comment:
Create a separate class for PickerView extension named OccasionPicker+Extensions. Also copy UICollectionViewDelegate, UICollectionViewDelegateFlowLayout, UIScrollViewDelegate, UICollectionViewDataSource

Reply:
Done